vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImpl.java

```diff
@@ -67,6 +68,9 @@
   private Set<FileUpload> fileUploads;
   private Session session;
   private User user;
+  private JsonArray parsedBodyAsJsonArray;
+  private JsonObject parsedBodyAsJson;
+  private Map<String, String> parsedBodyAsString;
```

Comment:
Isn't it an overkill to cache all encodings? Why not just cache the default encoding "null" and extract from the buffer other encodings? A new map here will just create gc as I guess most users will not switch encodings during a request that often, wdyt?

Reply:
@pmlopes I'm not sure I understand your point...  
I agree that on the same request context, the same encoding will probably be used (on most cases) over and over again. 
What do you suggest? removing the cache by encoding map?